django_forest/utils/schema.py

```diff
@@ -172,3 +179,37 @@ def handle_json_api_serializer(cls):
         for collection in cls.schema['collections']:
             # Notice: create marshmallow-jsonapi resource for json api serializer
             create_json_api_schema(collection)
+
+    @classmethod
+    def get_serialized_collection(cls, collection):
+        for index, field in enumerate(collection['fields']):
+            collection['fields'][index] = {x: field[x] for x in field if x in FIELD.keys()}
+        return collection
+
+    @classmethod
+    def handle_schema_file_production(cls, file_path):
+        try:
+            with open(file_path, 'r') as f:
+                data = f.read()
+                try:
+                    cls.schema_data = json.loads(data)
+                except Exception:
+                    logger.error('The content of .forestadmin-schema.json file is not a correct JSON.')
+                    logger.error('The schema cannot be synchronized with Forest Admin servers.')
+        except Exception:
+            logger.error('The .forestadmin-schema.json file does not exist.')
+            logger.error('The schema cannot be synchronized with Forest Admin servers.')
+
+    @classmethod
+    def handle_schema_file(cls):
+        file_path = os.path.join(os.getcwd(), '.forestadmin-schema.json')
+        if settings.DEBUG:
+            schema = copy.deepcopy(cls.schema)
+            for index, collection in enumerate(schema['collections']):
+                schema['collections'][index] = cls.get_serialized_collection(collection)
+
+            cls.schema_data = json.dumps(schema, indent=2)
+            with open(file_path, 'w') as f:
+                f.write(cls.schema_data)
+        else:
+            cls.handle_schema_file_production(file_path)
```

Comment:
This feels like:
1. Build and store schema (optional, dev only)
2. Read schema from storage

Having a `handle_schema_file_production` method made me think it got special treatment in production (or does it?)

Reply:
Yes, this is exactly the case when I'm looking to the `forest-rails` code: https://github.com/ForestAdmin/forest-rails/blob/39ff7f245ef1232985dd0aaa0d30c3b42e62cb40/lib/forest_liana/bootstrapper.rb#L73
A special treatment is made for the production environement. The `.forestadmin-schema.json` is always read even if the schema has been built before.
This is a bit strange to me... It is like what you send to the server is always the same thing, even if you modified things in your project BUT your project will live with a real in memory schema (built before).
Maybe there is something I'm missing regarding what you can do in production, and maybe DWO is a part of this?

Reply:
As far as I know, the schema is NOT modified by agents in non-dev environments. The idea is to have users modify things in dev, push their code and schema to git, and fetch the schema file from git on prod.

Schema is sent from production agent to forest server to update the apimap of the production env, and trigger rendering refresh.

DWO comes later in the process, while refreshing renderings.